core/base/ftmTree/FTMTreeUtils_Template.h

```diff
@@ -74,19 +65,247 @@ namespace ttk {
+                << " (" << std::get<1>(birthDeath) << ")" << std::endl;
+    }
 
-    template <typename type>
-    static type myAbs(const type var) {
-      return (var >= 0) ? var : -var;
+    template <class dataType>
+    void FTMTree_MT::printTreeScalars(bool printNodeAlone) {
+      std::streamsize ss = std::cout.precision();
+      for(unsigned int i = 0; i < this->getNumberOfNodes(); ++i) {
+        idNode iOrigin
+          = this->isNodeOriginDefined(i) ? this->getNode(i)->getOrigin() : i;
+        if(printNodeAlone
+           or (not printNodeAlone
+               and (not this->isNodeAlone(i)
+                    or not this->isNodeAlone(iOrigin))))
+          std::cout << i << " _ " << std::setprecision(12)
+                    << this->getValue<dataType>(i) << std::endl;
+      }
+      std::cout << " ------------------------------ " << std::endl;
+      std::cout.precision(ss);
     }
 
     template <class dataType>
-    bool isEqual(dataType first, dataType two, double eps = 1e-6) {
-      return myAbs<dataType>(first - two)
-             < eps * std::max(myAbs<dataType>(first), myAbs<dataType>(two));
+    std::stringstream FTMTree_MT::printPairsFromTree(bool useBD,
+                                                     bool printPairs,
+                                                     bool doPrint) {
+      std::stringstream ss;
+      std::vector<std::tuple<idNode, idNode, dataType>> pairs;
+      this->getPersistencePairsFromTree(pairs, useBD);
+      ss << "size=" << pairs.size() << std::endl;
+      if(printPairs)
+        for(auto pair : pairs) {
+          ss << std::get<0>(pair) << " ("
+             << this->getValue<dataType>(std::get<0>(pair)) << ") _ ";
+          ss << std::get<1>(pair) << " ("
+             << this->getValue<dataType>(std::get<1>(pair)) << ") _ ";
+          ss << std::get<2>(pair) << std::endl;
+        }
+      ss << " ------------------------------ " << std::endl;
+
+      if(doPrint)
+        std::cout << ss.str();
+      return ss;
+    }
+
+    template <class dataType>
+    void FTMTree_MT::printMultiPersPairsFromTree(bool useBD, bool printPairs) {
+      std::vector<std::tuple<idNode, idNode, dataType>> pairs;
+      this->getPersistencePairsFromTree(pairs, useBD);
+      std::vector<int> noOrigin(this->getNumberOfNodes(), 0);
+      int noMultiPers = 0;
+      for(auto pair : pairs) {
+        noOrigin[std::get<0>(pair)]++;
+        noMultiPers += (noOrigin[std::get<0>(pair)] > 1) ? 1 : 0;
+        noOrigin[std::get<1>(pair)]++;
+        noMultiPers += (noOrigin[std::get<1>(pair)] > 1) ? 1 : 0;
+      }
+      std::cout << "Number of multi pers pairs : " << noMultiPers << std::endl;
+      if(printPairs) {
+        auto multiPers = this->getMultiPersOrigins<dataType>(useBD);
+        for(auto node : multiPers)
+          std::cout << node << std::endl;
+      }
+      std::cout << " ------------------------------ " << std::endl;
```

Comment:
thanks for using `printMsg` instead (throughout)

Reply:
Ok! Done in the next commit.